adcc/ReferenceState.py

```diff
@@ -101,6 +101,12 @@ def __init__(self, hfdata, core_orbitals=None, frozen_core=None,
             parameter, the class will thus automatically import all ERI tensor
             and Fock matrix blocks.
 
+        gauge_origin : str or list, optional
+            Select the gauge origin for operator integrals.
+            Either by specifying a list in atomics units [x,y,z] or by choosing
+            one of the keywords (mass_center, charge_center, origin)
+            default: origin
+
```

Comment:
It feels like the gauge origin should rather be passed through the respective "property" functions, rather than through the `ReferenceState`. The current code needs to pass a lot of `kwargs` here and there, which is a bit "out of line" wrt the current code structure.
It can of course stay as a kwarg in `run_adc`. 

I suggest to implement sth like for the density-dependent solvent contributions:
Have the `gauge_origin` as `kwarg` for the respective property, i.e., `operators.nabla(gauge_origin=[0,0,0])`. Maybe one can also implement a shortcut that if one calls `operators.nabla` (i.e. as now), I'd have to look into that.